src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Export/GroupMemberExtractor.cs

```diff
@@ -0,0 +1,65 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Hl7.Fhir.Model;
+using Microsoft.Health.Extensions.DependencyInjection;
+using Microsoft.Health.Fhir.Core.Extensions;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.Core.Features.Search;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations.Export
+{
+    /// <summary>
+    /// Class to get member ids and types out of a group. Split between common and version specifc code due to the change between Stu3 and R4 to the ResourceReference object.
+    /// </summary>
+    public class GroupMemberExtractor : IGroupMemberExtractor
+    {
+        private readonly IScoped<IFhirDataStore> _fhirDataStore;
+        private readonly ResourceDeserializer _resourceDeserializer;
+        private readonly IReferenceToElementResolver _referenceToElementResolver;
+
+        public GroupMemberExtractor(
+            IScoped<IFhirDataStore> fhirDataStore,
+            ResourceDeserializer resourceDeserializer,
+            IReferenceToElementResolver referenceToElementResolver)
+        {
+            EnsureArg.IsNotNull(fhirDataStore, nameof(fhirDataStore));
+            EnsureArg.IsNotNull(resourceDeserializer, nameof(resourceDeserializer));
+            EnsureArg.IsNotNull(referenceToElementResolver, nameof(referenceToElementResolver));
+
+            _fhirDataStore = fhirDataStore;
+            _resourceDeserializer = resourceDeserializer;
+            _referenceToElementResolver = referenceToElementResolver;
+        }
+
+        public async Task<List<Tuple<string, string>>> GetGroupMembers(string groupId, CancellationToken cancellationToken)
+        {
+            var groupResource = await _fhirDataStore.Value.GetAsync(new ResourceKey(KnownResourceTypes.Group, groupId), cancellationToken);
+
+            var group = _resourceDeserializer.Deserialize(groupResource);
+            var groupContents = group.ToPoco<Group>().Member;
+
+            var members = new List<Tuple<string, string>>();
+
+            foreach (Group.MemberComponent member in groupContents)
```

Comment:
From my limited understanding of group, what happens if the `Group` resource does not define any members and only has `characteristic` defined? Do we have to handle that scenario by searching for resources based on the `characteristic` value?

Reply:
Good catch. This is an overlook of mine, I was only considering direct group members. I'll look into this.

Reply:
Added to https://microsofthealth.visualstudio.com/Health/_workitems/edit/74401

Reply:
Do you think it might be better to keep group export disabled until we support the above scenario also?